src/libraries/Microsoft.Extensions.Configuration.Binder/tests/Common/ConfigurationBinderTests.cs

```diff
@@ -193,13 +193,37 @@ public void EmptyStringIsNullable()
             var config = configurationBuilder.Build();
 
 #if BUILDING_SOURCE_GENERATOR_TESTS
-            Assert.Throws<NotSupportedException>(() => config.GetValue<bool?>("empty"));
+            // Ensure exception messages are in sync
+            Assert.Throws<InvalidOperationException>(() => config.GetValue<bool?>("empty"));
```

Comment:
I see this is because of parsing throwing Invalid Operation. why we cannot have the source gen return the same result as the runtime in such cases? I mean have it just return null.

Reply:
`NotSupportedException` was thrown by all the generated `GetValue` methods (introduced in [the last PR](https://github.com/dotnet/runtime/pull/85843)). `InvalidOperationException` is thrown now because the input string is not a valid bool. The reflection implementation is forgiving in this scenario (i.e. it returns `null`). I have a work item to get exception/format handling in sync, but I'm making it separate because there's quite a bit of nuance.

Reply:
Is there any reason we cannot keep the runtime behavior? I mean to have the source gen is forgiving too.

Reply:
We certainly can. Just that the parsing currently is funnels into methods like this: https://github.com/dotnet/runtime/blob/47cb253236789befdafdb430cd1c852ca98b1bf9/src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestPrimitivesGen.generated.txt#L270-L280

This exception handling is desired a lot of the time and would need to be adjusted to support these edge cases. It would be easiest to address this in isolation (probably after the other features - overloads, remaining collections, ctors - are implemented so that the full scope is being tested).

Reply:
Just to ensure I fully understand, are you saying this is a temporary behavior to throw such exception till we accommodate other features but will be changed before we ship to match the runtime behavior? or are you saying this is the behavior we'll stick with?

Reply:
> are you saying this is a temporary behavior to throw such exception till we accommodate other features but will be changed before we ship to match the runtime behavior?

Yes, indeed.